docs/modules/user-guide/partials/proc_using-the-devfile-library.adoc

```diff
@@ -0,0 +1,159 @@
+
+====
+----
+ // To get a slice of Kubernetes containers of type corev1.Container from the devfile component containers
+ containers, err := generator.GetContainers(devfile)
+
+ // To generate a Kubernetes deployment of type v1.Deployment
+ deployParams := generator.DeploymentParams{
+ 	TypeMeta:          generator.GetTypeMeta(deploymentKind, deploymentAPIVersion),
+ 	ObjectMeta:        generator.GetObjectMeta(name, namespace, labels, annotations),
+ 	InitContainers:    initContainers,
+ 	Containers:        containers,
+ 	Volumes:           volumes,
+ 	PodSelectorLabels: labels,
+ }
+ deployment := generator.GetDeployment(deployParams)
+----
+====
+
+* Update the devfile content by running:
+
+====
+----
+// To update an existing component in the devfile object
+err := devfile.Data.UpdateComponent(v1.Component{
+	    Name: "component1",
+	    ComponentUnion: v1.ComponentUnion{
+	    	Container: &v1.ContainerComponent{
+	    		Container: v1.Container{
+	    			Image: "image1",
+             },
+         },
+     },
+})
+
+// To add a new component to the devfile object
+err := devfile.Data.AddComponents([]v1.Component{
+     {
+        Name: "component2",
+        ComponentUnion: v1.ComponentUnion{
+            Container: &v1.ContainerComponent{
+                Container: v1.Container{
+                    Image: "image2",
+                },
+            },
+        },
+     },
+})
+
+// To delete a component from the devfile object
+err := devfile.Data.DeleteComponent(componentName)
+----
+====
+
+* Write a devfile by running:
+
+====
+----
+// If the devfile object has been created with the devfile path already set, can simply call WriteYamlDevfile to write the devfile
+err := devfile.WriteYamlDevfile()
+
+
+// To write to a devfile from scratch
+// create a new DevfileData with a specific devfile version
+devfileData, err := data.NewDevfileData(devfileVersion)
+
+// set schema version
+devfileData.SetSchemaVersion(devfileVersion)
+
+// To add devfile content, use library APIs
+devfileData.AddComponents([]v1.Component{...})
+devfileData.AddCommands([]v1.Commands{...})
+......
+
+// create a new DevfileCtx
+ctx := devfileCtx.NewDevfileCtx(devfilePath)
+err = ctx.SetAbsPath()
+
+// create devfile object with the new DevfileCtx and DevfileData
+devfile := parser.DevfileObj{
+ 	Ctx:  ctx,
+ 	Data: devfileData,
+}
+
+// write to the devfile on disk
+err = devfile.WriteYamlDevfile()
+----
+====
+
+.Additional resources
+
+* For more information on parsing a devfile, see the link:https://github.com/devfile/library/blob/main/pkg/devfile/parse.go[parse.go source file].
+* For more information on generating Kubernetes objects with a devfile, see the link:https://github.com/devfile/library/blob/main/pkg/devfile/generator/generators.go[generators.go source file].
+* For information on validating a devfile, see the link:https://github.com/devfile/library/blob/main/pkg/devfile/validate/validate.go[writer.go source file].
```

Comment:
```suggestion
* For information on validating a devfile, see the link:https://github.com/devfile/library/blob/main/pkg/devfile/validate/validate.go[validate.go source file].
```

Reply:
also information on writing a devfile?

Reply:
Resolving this conversation since the RH guidelines don't want us to link to external github pages.